### ISSUE TITLE
[r2] -d + pxl + /R/ [..]

### ISSUE THREAD

**Issue by zonkzonk** [radare/radare2#1944]
morrn,

some crashes when combining -d, /R/ and pxl

```
tail -2 todo-fuzz-jmp 
r2 -b 64 -e asm.stackptr=true -e cmd.hit=pxl -e search.maxhits=2 -dqc 
"/R/ jmp rip+0x7f0ceb2efdc0" a.out.bak
r2 -b 64 -e asm.stackptr=true -e cmd.hit=pxl -e search.maxhits=2 -dqc 
"/R/ jmp 0xffffffffffffffff+rip" a.out.bak
```

```

segfault at 7f10edafbfdc ip 00007f114b9d2a20 sp 00007fff154fd8a0 error 4 in libr_asm.so.0.9.9-git[7f114b83c000+3b8000]
segfault at 7fa1e8e65fdc ip 00007fa242d8ca20 sp 00007fff6809bed0 error 4 in libr_asm.so.0.9.9-git[7fa242bf6000+3b8000]

,ls -lh core*
-rw------- 1 zlul lulzgroup 977M Jan 12 00:08 core.4608
-rw------- 1 zlul lulzgroup 6.8M Jan 12 00:12 core.4682

Core was generated by `r2 -b 64 -e asm.stackptr=true -e cmd.hit=pxl -e search.maxhits=2 -dqc /R/ jmp r'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f114b9d2a20 in ?? ()
(gdb) bt
#0  0x00007f114b9d2a20 in ?? ()
Backtrace stopped: Cannot access memory at address 0x7fff154fd8a0
(gdb) i r
rax            0x7f10edafbfdc   139710683922396
rbx            0x21b90f0        35361008
rcx            0x5b59a273       1532600947
rdx            0x7f10edafbfdc   139710683922396
rsi            0x7fff154fda30   140733550942768
rdi            0x2182440        35136576
rbp            0x7fff154fd8d0   0x7fff154fd8d0
rsp            0x7fff154fd8a0   0x7fff154fd8a0
r8             0xffffffff       4294967295
r9             0x228c3c0        36225984
r10            0x0      0
r11            0x5      5
r12            0x4028f0 4204784
r13            0x7fff154ff410   140733550949392
r14            0x0      0
r15            0x0      0
rip            0x7f114b9d2a20   0x7f114b9d2a20
eflags         0x10202  [ IF RF ]
cs             0x33     51
ss             0x2b     43
ds             0x0      0
es             0x0      0
fs             0x0      0
gs             0x0      0
(gdb) x/i $pc
=> 0x7f114b9d2a20:      Cannot access memory at address 0x7f114b9d2a20
(gdb) i f
Stack level 0, frame at 0x7fff154fd8a8:
 rip = 0x7f114b9d2a20; saved rip = <error: Cannot access memory at address 0x7fff154fd8a0>
 Outermost frame: Cannot access memory at address 0x7fff154fd8a0
 Arglist at 0x7fff154fd898, args: 
 Locals at 0x7fff154fd898, Previous frame's sp is 0x7fff154fd8a8
Cannot access memory at address 0x7fff154fd8a0
```

r2 -v
377ec87272617e3804ed04d5a3f1575f1e6b7499

Greetings
--zlul

**Comment by crowell** [comment]
@zonkzonk can you provide the a.out.bak pls?

**Comment by zonkzonk** [comment]
@crowell http://sprunge.us/MVjU

**Comment by crowell** [comment]
the `pxl` stuff has nothing to do with it, just the fact that searching for gadgets under the debugger searches in places that are not accessable

**Comment by zonkzonk** [comment]
ok, then we can simplify to r2 -b 64 -e asm.stackptr=true  -e search.maxhits=2 -dqc 
"/R/ pop" a.out.bak

**Comment by zonkzonk** [comment]
this is fixed in 64bit 8f7fcc91cebb183ea62833a2c733a11ebed7a8a6 but not in 32bit (with wrong -b on purpose)

**Comment by zonkzonk** [comment]
with @crowell s patch:

```
git diff
diff --git a/libr/cons/grep.c b/libr/cons/grep.c
index cb86d1d..38bde6e 100644
--- a/libr/cons/grep.c
+++ b/libr/cons/grep.c
@@ -99,6 +99,7 @@ R_API void r_cons_grep(const char *str) {
        }
        ptr = buf;
        ptr3 = strchr (ptr, '['); // column number
+        cons->num->callback = NULL;
        if (ptr3) {
                ptr3[0] = '\0';
                cons->grep.tokenfrom = r_num_get (cons->num, ptr3+1);
``` 

it is  fixed on 32 bit:

```
zlul@debian:~/bins$ r2 -b 64 -e asm.stackptr=true -e search.maxhits=2 -dqc "/R/ pop" a.out.bak 
Process with PID 21280 started...
PID = 21280
pid = 21280 tid = 21280
r_debug_select: 21280 21280
Using BADDR 0x8048000
Asuming filepath ./a.out.bak
bits 64
pid = 21280 tid = 21280
zlul@debian:~/bins$ ./a.out.bak: 2: ./a.out.bak: Syntax error: "(" unexpected
```


 and 64bit: 

```
r2 -b 64 -e asm.stackptr=true -e search.maxhits=2 -dqc "/R/ pop" a.out.bak 
Process with PID 358 started...
PID = 358
pid = 358 tid = 358
r_debug_select: 358 358
Using BADDR 0x400000
bits 32
pid = 358 tid = 358
./a.out.bak: error while loading shared libraries: libr_util.so.0.9.7git: cannot open shared object file: No such file or directory
``` 


standard rop for pop takes:

```
32bit
real	0m5.937s
user	0m2.772s
sys	0m0.008s

64bit
real	0m8.106s
user	0m5.310s
sys	0m0.027s
```

with -d it takes however long
